exchanges/gateio/gateio_convert.go

```diff
@@ -0,0 +1,411 @@
+		}
+	}
+	if child.QuoteVolume != "" {
+		if a.QuoteVolume, err = strconv.ParseFloat(child.QuoteVolume, 64); err != nil {
+			return err
+		}
+	}
+	if child.High24H != "" {
+		if a.High24H, err = strconv.ParseFloat(child.High24H, 64); err != nil {
+			return err
+		}
+	}
+	if child.Low24H != "" {
+		if a.Low24H, err = strconv.ParseFloat(child.Low24H, 64); err != nil {
+			return err
+		}
+	}
+	if child.LowestAsk != "" {
+		if a.LowestAsk, err = strconv.ParseFloat(child.LowestAsk, 64); err != nil {
+			return err
+		}
+	}
+	if child.HighestBid != "" {
+		if a.HighestBid, err = strconv.ParseFloat(child.HighestBid, 64); err != nil {
+			return err
+		}
+	}
+	if child.EtfLeverage != "" {
+		if a.EtfLeverage, err = strconv.ParseFloat(child.EtfLeverage, 64); err != nil {
+			return err
+		}
+	}
+	if child.Last != "" {
+		if a.Last, err = strconv.ParseFloat(child.Last, 64); err != nil {
+			return err
+		}
+	}
+	a.EtfPreTimestamp = time.Unix(child.EtfPreTimestamp, 0)
+	return nil
+}
+
+// UnmarshalJSON to deserialize timestamp information and create OrderbookItem instance from the list of asks and bids data.
+func (a *OptionsTicker) UnmarshalJSON(data []byte) error {
+	type Alias OptionsTicker
+	chil := &struct {
+		*Alias
+		LastPrice             string `json:"last_price"`
+		MarkPrice             string `json:"mark_price"`
+		IndexPrice            string `json:"index_price"`
+		MarkImpliedVolatility string `json:"mark_iv"`
+		BidImpliedVolatility  string `json:"bid_iv"`
+		AskImpliedVolatility  string `json:"ask_iv"`
+		Leverage              string `json:"leverage"`
+	}{
+		Alias: (*Alias)(a),
+	}
+	err := json.Unmarshal(data, chil)
+	if err != nil {
+		return err
+	}
+	if chil.LastPrice != "" {
+		a.LastPrice, err = strconv.ParseFloat(chil.LastPrice, 64)
+		if err != nil {
+			return err
+		}
+	}
+	if chil.MarkPrice != "" {
+		a.MarkPrice, err = strconv.ParseFloat(chil.MarkPrice, 64)
+		if err != nil {
+			return err
+		}
+	}
+	if chil.IndexPrice != "" {
+		a.IndexPrice, err = strconv.ParseFloat(chil.IndexPrice, 64)
+		if err != nil {
+			return err
+		}
+	}
+	if chil.MarkImpliedVolatility != "" {
+		a.MarkImpliedVolatility, err = strconv.ParseFloat(chil.MarkImpliedVolatility, 64)
+		if err != nil {
+			return err
+		}
+	}
+	if chil.BidImpliedVolatility != "" {
+		a.BidImpliedVolatility, err = strconv.ParseFloat(chil.BidImpliedVolatility, 64)
+		if err != nil {
+			return err
+		}
+	}
+	if chil.AskImpliedVolatility != "" {
+		a.AskImpliedVolatility, err = strconv.ParseFloat(chil.AskImpliedVolatility, 64)
+		if err != nil {
+			return err
+		}
+	}
+	if chil.Leverage != "" {
+		a.Leverage, err = strconv.ParseFloat(chil.Leverage, 64)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// UnmarshalJSON to deserialize timestamp information and create OrderbookItem instance from the list of asks and bids data.
+func (a *Orderbook) UnmarshalJSON(data []byte) error {
+	type Alias Orderbook
+	type askorbid struct {
+		Price string  `json:"p"`
```

Comment:
```suggestion
		Price float64 `json:"p,string"`
```

Skips having to do the ParseFloat logic below